toggl/mobileapp
# Improve GroupCollectionViewSource

Closes #1523 

This PR:

- Wraps the animations in Begin/EndUpdates blocks, which is recommended by Apple
- Handlles all cases of NotifyCollectionChangedAction and defaults to TableView.ReloadData()
- Removes the try/catch blocks from the animations, since they aren’t really catching anything (the throw happens asynchronously in the TableView, no in the Source)
- Puts a lock around the switch that tries to do the animations 

REPRO STEPS FOR TESTING:
- Open app
- Start new time entry
- Ensure it’s synced and running
- Put the app into the background
- Go to the web app, stop the running time entry
- Create a bunch of tes in the web app
- Start a new te in the web app and leave it running
- Wait for 5 minutes, the time it takes for the background service to kick in (alternatively, reduce the time it takes for it to kick in by changing the value in FoundationExtensions.cs, line 139)
- Open the app and it should not crash

Test with new users (welcome view), old users with no TEs (spider view) and also with users that have recent time entries.

:bug: :gun:
